packages/slice-machine/src/features/onboarding/useOnboarding.ts

```diff
@@ -0,0 +1,70 @@
+import { OnboardingStepId } from "@prismicio/editor-fields";
+import { updateData, useRequest } from "@prismicio/editor-support/Suspense";
+
+import { managerClient } from "@/managerClient";
+
+async function getOnboarding() {
+  try {
+    return await managerClient.prismicRepository.fetchOnboarding();
+  } catch (e) {
+    console.error("Error while trying to get onboarding", e);
+    return undefined;
+  }
+}
+
+export function useOnboarding() {
+  const onboarding = useRequest(getOnboarding, []);
+
+  async function toggleStep(stepId: OnboardingStepId) {
+    if (!onboarding) return [];
+
+    const previousSteps = onboarding.completedSteps;
+
+    const newSteps = previousSteps.includes(stepId)
+      ? previousSteps.filter((id) => id !== stepId)
+      : [...previousSteps, stepId];
+
+    updateData(getOnboarding, [], {
```

Comment:
💡Shouldn't you update the data only after it's a success in the back-end?

It would prevent to revert the completed steps after and risk a flash change.
This would even prevent to have the `newSteps` definition, I guess that is error-prone, no?

Reply:
that was initial version, but thought that this optimistic update is better, since it take some time to get the response from the service, and then there is quite a delay between marking the step as done and having this reflected in ui.

Reply:
Did the same as in editor, since in this particular case, we want to avoid having a delay before throwing the confetti after completing the last step, this mutation doesn't have an optimistic update anymore ✅